lib/stream_data.ex

```diff
@@ -701,20 +722,48 @@ defmodule StreamData do
   #       |> LazyTree.flatten()
   #     end)
   #
-  @spec list_of(t(a)) :: t([a]) when a: term
-  def list_of(data) do
+  @spec list_of(t(a), keyword) :: t([a]) when a: term
+  def list_of(data, options \\ []) do
+    list_length_range_fun = list_length_range_fun(Keyword.take(options, [:length, :min_length, :max_length]))
+
     new(fn seed, size ->
       {seed1, seed2} = split_seed(seed)
-      length = uniform_in_range(0..size, seed1)
+      min_length.._ = length_range = list_length_range_fun.(size)
+      length = uniform_in_range(length_range, seed1)
 
       data
       |> call_n_times(seed2, size, length, [])
       |> LazyTree.zip()
-      |> LazyTree.map(&list_lazy_tree/1)
+      |> LazyTree.map(&list_lazy_tree(&1, min_length))
       |> LazyTree.flatten()
     end)
   end
 
+  defp list_length_range_fun(options) do
+    {min, max} =
+      case options[:length] do
```

Comment:
Yeah yeah, problem is that we still allow it for `:min_length` and `:max_length` :|